crates/ide-diagnostics/src/tests/regressions.rs

```diff
@@ -0,0 +1,29 @@
+//! Regression tests that don't belong to any specific handler but still require full
```

Comment:
I feel like it's better to put this under `crates/ide-diagnostics/src/handlers/missing_match_arms.rs` so it's clear what kind of diagnostics this test tests.